### PR TITLE
Updating with model the StructuralMechanics main script

### PR DESCRIPTION
**Pull request by loumalouomega**
Concerning #1045

### CODE REVIEW

File: applications/StructuralMechanicsApplication/python_scripts/kratos_main_structural.py [hunk 1]
```diff
@@ -51,12 +52,12 @@
     gid_output.ExecuteInitialize()
 
 ## Creation of the Kratos model (build sub_model_parts or submeshes)
-StructureModel = {ProjectParameters["problem_data"]["model_part_name"].GetString(): main_model_part}
-
-## Get the list of the sub_model_parts in where the processes are to be applied
+StructureModel = Model()
+StructureModel.AddModelPart(main_model_part)
+# Get the list of the sub_model_parts in where the processes are to be applied
 for i in range(ProjectParameters["solver_settings"]["processes_sub_model_part_list"].size()):
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/python_scripts/kratos_main_structural.py]: you don't need to do this. Once you add the StructureModel, it adds alone all of its submodelparts. That's actually the point of it... (one of)

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/kratos_main_structural.py]: The problem is to keep to retrocompatibilty with what is done, this is the only way

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/kratos_main_structural.py]: I know the submodelparts are added automatically, but if you want to do it that way we need to change all processes, that can be done in the future, but is a lot of work

**Comment by RiccardoRossi** [comment]
no!

mpdel defines the operator and it has an identical onterface. just try it
out

Riccardo

**Comment by loumalouomega** [comment]
I saw the interface at the model_test.py, by default it adds the submodelparts as for example:

`model["main.inlet"]` but our current implementation for the processes works as `model["inlet"]`

**Comment by RiccardoRossi** [comment]
the model provides both the flattened interface

     model["inlet"]

and the "." interface
 
    model["main.inlet"]

**Comment by loumalouomega** [comment]
[test.gid.tar.gz](https://github.com/KratosMultiphysics/Kratos/files/1488961/test.gid.tar.gz)

**Comment by loumalouomega** [comment]
imagen
imagen

**Comment by loumalouomega** [comment]
imagen

imagen

As you can see you need to add the name of the main to access the submodelpart, then we need to to this tricky thing in order to get the submodelpart, if you want to proceed in a different way we need to change ALL python processes

**Comment by loumalouomega** [comment]
@RiccardoRossi I added the update to keep consistent the PR, I can revert it

**Comment by loumalouomega** [comment]
Ping

**Comment by RiccardoRossi** [comment]
i think this change is for good, and i would do the same for the FluidDynamics main. however let's see the opinion of @KratosMultiphysics/technical-committee

**Comment by loumalouomega** [comment]
The project parameters can be simplified with this change

**Comment by philbucher** [comment]
Question:
After this is done, all the submodelparts of the mainmodelpart are in the Model?

**Comment by loumalouomega** [comment]
Yes

El 27 nov. 2017 9:32 a. m., "Philipp Bucher" <notifications@github.com>
escribió:

Question:
After this is done, all the submodelparts of the mainmodelpart are in the
Model?

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub
<https://github.com/KratosMultiphysics/Kratos/pull/1053#issuecomment-347111308>,
or mute the thread

**Comment by RiccardoRossi** [comment]
all the submodelparts that were in the model_part at the moment in which you add it to the model.

the problem is that if you generate a new submodelpart after that moment it will NOT be added automatically. This is something that we eventually want to change, so that every time a new model part is created it is automatically added to the model. However this is not there right now...and it is the reason for which the Model interface is subjected to changes in the future

**Comment by philbucher** [comment]
Thats cool! (Bcs a process is constructed with a Model and then we don't have to think abt whether a smp is there or not)

**Review by maceligueta — APPROVED** [review]

**Pull request merged**